code/src/java/pcgen/rules/context/AbstractReferenceContext.java

```diff
@@ -127,13 +128,17 @@ public boolean validate(UnconstructedValidator validator)
 		return getManufacturer(c, cat).getTypeReference(val);
 	}
 
-	public <T extends Loadable> T constructCDOMObject(Class<T> c, String val)
+	public <T extends Loadable, U extends Loadable & CategorizedCDOMObject<U>> T constructCDOMObject(Class<T> c, String val)
```

Comment:
Could you add a comment on this method please? I'm still uncertain what U is and how it affects things.